lib/ddtrace/span_operation.rb

```diff
@@ -189,6 +182,22 @@ def finished?
       span.stopped?
     end
 
+    # Set this span's parent, inheriting any properties not explicitly set.
+    # If the parent is nil, set the span as the root span.
+    def parent=(parent)
+      @parent = parent
+
+      if parent.nil?
+        span.trace_id = span.span_id
+        span.parent_id = 0
+      else
```

Comment:
Minor: I wonder if it'd make sense to make this more explicit. E.g. have a deliberate and explicit `make_root_span` method if one really wants to make a span into a root, and otherwise complain if someone tries to do it via setting `parent = nil`.

Reply:
Yes, this method's API is super convoluted.
I think we actually want to remove it altogether, specially this special `nil` case.
I'll document this.